### ISSUE TITLE
Some command line output ignores system foreground color configuration

### ISSUE THREAD

**Issue by jrob** [cake-build/cake#1633]
### What You Are Seeing?

When running in a shell with a white background, some text is invisible.

Screenshot:

![screenshot](https://user-images.githubusercontent.com/613402/26934881-8fe675d8-4c38-11e7-9dfa-d5954286f9f2.png)

Actual text:

```
$ ./build.ps1 --showdescription
Preparing to run build script...
Running build script...
Analyzing build script...
Processing build script...
Installing tools...
Installing addins...
Compiling build script...

Task                          Description
===============================================================================
Restore-NuGet-Packages
```

### What is Expected?

The output should respect the configured foreground colors so that text is visible.

### What version of Cake are you using?

` Version 0.19.5+Branch.main.Sha.e4c4ad2a949637f89063dc7a90363839b740af2c`

### Are you running on a 32 or 64 bit system?

64

### What environment are you running on?  Windows? Linux? Mac?

Windows 10 with the default shell.

### How Did You Get This To Happen? (Steps to Reproduce)

Current colors set for the shell

~~~ps
$colors = [Enum]::GetValues( [ConsoleColor] )
$max = ($colors | foreach { "$_ ".Length } | Measure-Object -Maximum).Maximum
foreach( $color in $colors ) {
    Write-Host (" {0,2} {1,$max} " -f [int]$color,$color) -NoNewline
    Write-Host "$color" -Foreground $color
}
~~~

![CLI Colors](https://cloud.githubusercontent.com/assets/613402/21781401/266764a4-d67d-11e6-8cdf-4150347cf242.PNG)

Windows powershell properties. 

![Powershell Properties](https://cloud.githubusercontent.com/assets/613402/21781617/3f9738d6-d67e-11e6-8bb4-f455bfd3374e.PNG)

**Comment by mcraiha** [comment]
![hardtoread](https://user-images.githubusercontent.com/11350236/72912454-5452bb80-3d44-11ea-8f48-3b4ef7827388.png)
With macOS default foreground color is hard to read when terminal is running with white background color

**Comment by augustoproiete** [comment]
@jrob, this has been fixed with PR https://github.com/cake-build/cake/pull/3094 and will be released with Cake v1.1.0

@mcraiha The current color palette of Cake doesn't work well with light backgrounds. We have an issue open to [support custom ConsolePalette/ANSI colors](https://github.com/cake-build/cake/issues/2976) which eventually would provide an alternative for the issue you pointed out.

Another alternative which will be available in the next release of Cake (v1.1.0 as of this writing) is the [ability to disable colors with an environment variable called `NO_COLOR`](https://github.com/cake-build/cake/issues/2975) which would be a workaround for devs using light backgrounds until #2976 or something better comes along.
